rfcs/trust-and-safety/20230109-trust_and_safety_preliminary_overview_and_exploration.md

```diff
@@ -0,0 +1,285 @@
+1. As an Openverse user, I want to know why a result is marked as "sensitive"
+   without needing to view the result itself so that I can make an informed
+   decision about whether a sensitive (but not policy-violating) result is
+   relevant to what I am searching for
+1. As an Openverse content moderator, I want to be able to bulk remove results
+   from a given search so that a search that has mixed material (results that
+   follow and results that do not follow Openverse's content policies) can be
+   easily dealt with
+1. As an Openverse content moderator, I want to be able to remove all results
+   from a particular creator so that if a problematic creator is discovered and
+   all (or practically all, or mostly all) their content does not follow
+   Openverse's content policies, they can be easily removed
+1. As an Openverse content moderator, I want to be able to mark all results from
+   a creator as needing review so that if a problematic creator is discovered we
+   don't need to manually mark each result for review (or removal, depending on
+   the severity of the creator's issues)
+1. As an Openverse content moderator, I want to be able to upstream moderation
+   decisions to providers when appropriate so that Openverse can be a good
+   steward of the commons and also help improve the quality of providers'
+   content
+1. As an Openverse user, I expect that Openverse will never include material
+   already known to be illegal because Openverse is not a place for distribution
+   of illegal materials
+1. As an Openverse content moderator, I expect that results with sensitive
+   textual material will be automatically raised for review so that I can
+   quickly identify results that may need moderation action without a user
+   needing to file a report or a content moderator searching them out
+1. As an Openverse user, I expect that sensitive textual material has a useful
+   content warning so that I can make an informed decision about whether I want
+   to read a piece of text
+1. As an Openverse user, I expect to be able to clearly understand what content
+   on the Openverse website is from Openverse and what content is from the
+   providers so that I understand whether I need to make upstream reports of
+   creators or specific content to providers or to Openverse
+1. As a creator, I want to be able to obtain the rationale behind one or many of
+   my works being removed from Openverse
+1. As a creator, I want to be able to mark my own content with specific content
+   warnings so that I can make my content more accessible
+1. As a user, I want to be able to mark or suggest content warnings for results
+   so that I can help make Openverse catalogued content more accessible
+
+- Assumptions:
+  - Upon reingestion from a provider, the catalogue is able to note when results
+    have been removed from upstream providers and eagerly remove them from the
+    Openverse API without needing a full data refresh to occur
```

Comment:
Does this mean that we want to remove all the items that were not returned from the provider API when reingesting? This might mean that we remove a lot of items if the provider changes the API or if the API errors out for any reason. However, re-checking each individual item that wasn't present during reingestion also requires time and resources.

I wonder if adding some sort of scanning process for items that are not present at the provider when re-ingesting would help. Or we could also consider adding an "antitboosting" parameter (is there a word for the action that is opposite of search rank boosting?) to all of such items, assuming that they were removed from the provider for some reason.

Reply:
Oh you're right. Even from a non-technical perspective, it's a difficult question. Expanding on your list, I can think of a bunch of reasons why a work might not appear in the provider API anymore, and there's almost no way for us to distinguish between them.

1. The creator deletes their account or an individual validly licensed work. In this case, the works are still CC licensed (assuming they were correctly licensed to begin with, i.e., not stolen) and part of the commons and would ideally still be accessible. We've discussed this a couple of times in the past and mused about whether Openverse has a role in "preserving" the commons somehow (maybe by uploading things to Archive.org or something like that).
2. DMCA takedown. We shouldn't distribute these.
3. Sensitive material takedown. We shouldn't distribute these either _probably_ assuming we agree with the provider. For example, the provider could be participating in government censorship that we don't care to be a part of, do we preserve these as in the first case then, assuming they're correctly licensed CC works?
4. Illegal material takedown. We shouldn't distribute these.
5. API errors (as you mentioned).
6. API changes (also as you mentioned).

Each of these are different, and I don't think we could easily tell the difference between any of them in an automated or even manual way without heavy provider involvement.

One thing to note that I forgot about when writing this is that we will eventually stop serving those results in search because the links will be dead. They'll disappear from search after the cached success response expires (30 days from the first appearance in search). The thumbnail will continue to exist though, and I think the single result will as well.

Complicated issue. I don't have any concrete suggestions for this at the moment.

> I wonder if adding some sort of scanning process for items that are not present at the provider when re-ingesting would help. Or we could also consider adding an "antitboosting" parameter (is there a word for the action that is opposite of search rank boosting?) to all of such items, assuming that they were removed from the provider for some reason.

Boosting them downward is an interesting idea. You might be able to apply negative or fractional boosting scores to documents in ES that would cause their scores to plummet. At that point though, I wonder if we could "soft delete" them by setting a flag that would just exclude them from search? Maybe we should just play it safe and exclude them from single results as well for now until we have clearer understandings of what the alternatives and implications of those alternatives would be?

Very tricky issue!

Reply:
One discussed approach to this was to create `{mediaType}-removed` tables in the Catalog DB in which we insert any records removed from the main tables. We could match the schema of the main tables but also add `removed_on` and `removed_reason` columns which would indicate the date of removal and the reason for removal. As discussed, we likely can't always find the specific reason why an image was removed/became unavailable from the provider (although perhaps _some_ providers return different error codes in different situations, as an example, and we could leverage that) but this column would also be used for media _we_ remove from Openverse (for various content safety and copyright reasons).

This would allow us to create a DAG or other mechanism in the future to re-crawl the items in the `-removed` tables, either all of them or only ones with specific values in the `removed_reason` columns (for example we could only re-crawl items that 503 errored during the data refresh).

Reply:
Sounds great to me.